clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -551,24 +541,21 @@ getHLSLResourceAttrFromEitherDecl(VarDecl *VD,
 
     // the resource attr could be on the record decl itself or on one of
     // its fields (the resource handle, most commonly)
```

Comment:
I was going to add a comment about the name of this function, but then I realized I'm not sure that we really want it to be as flexible as it is.  I think we actually only want to support the attribute being on the handle member, we don't need to support it being on the record itself, do we?  I think we should be crisp about this rather than being flexible - it'll reduce the testing coverage needed as well.

Reply:
Yes, I think I may have left support for that as we transitioned moving the attr to the handles, but now it's no longer needed.